config/defines.inc.php

```diff
@@ -24,9 +24,15 @@
  * International Registered Trademark & Property of PrestaShop SA
  */
 
+$debugIps = require __DIR__.DIRECTORY_SEPARATOR.'debug_ip.php';
+
 /* Debug only */
 if (!defined('_PS_MODE_DEV_')) {
-    define('_PS_MODE_DEV_', true);
+    if (in_array($_SERVER['REMOTE_ADDR'], $debugIps)) {
```

Comment:
Not the best way to get client IP address.

Reply:
Maintenance mode uses Tools::getRemoteAddr()

Reply:
It can't be used because the autoloader is not loaded yet :thinking: